src/loader/playlist-loader.ts

```diff
@@ -240,7 +240,7 @@ class PlaylistLoader implements NetworkComponentAPI {
 
     // Override level/track timeout for LL-HLS requests
     // (the default of 10000ms is counter productive to blocking playlist reload requests)
-    if (context.deliveryDirectives?.part) {
+    if (context.deliveryDirectives?.part !== null) {
```

Comment:
Hi @jhonalino,

Thanks for the contribution!

This statement can evaluate to null or undefined based on `part` being optional and undefined 
https://github.com/video-dev/hls.js/blob/3521bd541f7645b5e5ca4b7eb592c27d56d435f4/src/types/level.ts#L62-L64
or `deliveryDirectives?` being either undefined or null.
https://github.com/video-dev/hls.js/blob/3521bd541f7645b5e5ca4b7eb592c27d56d435f4/src/types/loader.ts#L191

One way to only proceed only if `part` is a usable number is to check if it is finite:
 
```suggestion
    if (Number.isFinite(context.deliveryDirectives?.part)) {
```

Reply:
oh I did not know about that Number.isFinite @robwalch that is good to know! and thank you so much for taking a look at this.